### PR TITLE
Add more filters for admin notification targeting

### PR DESCRIPTION
**Pull request by knolleary**
## Description

Allows notification hub messages to be targeted to users of certain team types and billing states.

image

### CODE REVIEW

File: forge/routes/api/admin.js [hunk 1]
```diff
@@ -449,18 +449,26 @@ module.exports = async function (app) {
...
+            const count = await app.db.models.User.byTeamRole(recipientRoles, { teamTypes, billing, summary: true, count: true })
             reply.send({
                 mock: true,
                 recipientCount: count
             })
             return
         }
 
-        const recipients = await app.db.models.User.byTeamRole(recipientRoles, { summary: true })
+        const recipients = await app.db.models.User.byTeamRole(recipientRoles, { teamTypes, summary: true })
```
Comment (cstns) [inline forge/routes/api/admin.js:471]:
```suggestion
        const recipients = await app.db.models.User.byTeamRole(recipientRoles, { teamTypes, billing, summary: true })
```

File: frontend/src/pages/admin/NotificationsHub.vue [hunk 2]
```diff
@@ -50,78 +82,133 @@
...
+                            text: 'Your filters matched no users.',
+                            confirmLabel: 'Cancel',
+                            canBeCanceled: false
+                        })
+                    } else {
+                        Dialog.show({
+                            header: 'Platform Wide Announcement',
+                            kind: 'danger',
+                            text: `You are about to send an announcement to ${mockRes.recipientCount} ${pluralize('user', mockRes.recipientCount)}.`,
+                            confirmLabel: 'Continue'
```
Comment (cstns) [inline frontend/src/pages/admin/NotificationsHub.vue:179]:
```suggestion
                            confirmLabel: 'Continue',
                            canBeCanceled: false
```

attempting to send notifications to 0 users, refining parameters and sending a mock response caused the dialog to pop up without the possibility of canceling (i had to refresh the page)

**Review by cstns — CHANGES_REQUESTED** [review]

File: forge/routes/api/admin.js (hunk 1 above)
Comment (cstns) [reply forge/routes/api/admin.js:471]: I assume billing was meant to be passed along to the subsequent call

File: frontend/src/pages/admin/NotificationsHub.vue (hunk 2 above)
Comment (knolleary) [reply frontend/src/pages/admin/NotificationsHub.vue:179]: Odd - sounds like a bug with the Dialog service if a previous call can influence a future call.

File: forge/routes/api/admin.js (hunk 1 above)
Comment (knolleary) [reply forge/routes/api/admin.js:471]: Hmm, yes. Let me revisit as my testing locally definitely covered this...

File: frontend/src/pages/admin/NotificationsHub.vue (hunk 2 above)
Comment (knolleary) [reply frontend/src/pages/admin/NotificationsHub.vue]: Ooops - this should be set to `true` not `false`

Comment (cstns) [reply frontend/src/pages/admin/NotificationsHub.vue]: my bad!

**Pull request merged**